server/src/db/migrations/20210805190441_createBoxes.cjs

```diff
@@ -0,0 +1,23 @@
+/**
+ * @typedef {import("knex")} Knex
+ */
+
+/**
+ * @param {Knex} knex
+ */
+ exports.up = async (knex) => {
+  return knex.schema.createTable("boxes", (table) => {
+    table.bigIncrements("id")
+    table.text("allBoxes").notNullable()
+    table.bigInteger("puzzleId").notNullable().unsigned().index().references("puzzles.id")
+    table.timestamp("createdAt").notNullable().defaultTo(knex.fn.now())
+    table.timestamp("updatedAt").notNullable().defaultTo(knex.fn.now())
+  })
+}
```

Comment:
lets sync again. If you are formally moving to a compound data structure as we've been talking about this week, you won't even need a `boxes` table. Just a `puzzles` table with a `puzzleStructure` column that contains the array of arrays. You can store an array of arrays in a database as a JSON string. 
the `.json` column in Knex can help you with this. You just need to convert your array of arrays into a JSON string before adding it to your database